### PR TITLE
Handle nodestatus change

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: umcontroller/umcontroller.go [hunk 1]
```diff
@@ -857,6 +845,17 @@ func (umCtrl *Controller) createStateMachine() (string, []fsm.EventDesc, map[str
 			{Name: evUpdateStatusUpdated, Src: []string{stateStartRevert}, Dst: stateUpdateUmStatusOnRevert},
 			{Name: evContinue, Src: []string{stateUpdateUmStatusOnRevert}, Dst: stateStartRevert},
 			{Name: evSystemReverted, Src: []string{stateStartRevert}, Dst: stateIdle},
+			// process close connection
+			{Name: evConnectionClose, Src: []string{stateIdle}, Dst: stateInit},
```
Comment (mlohvynenko) [inline umcontroller/umcontroller.go:849]: why not  
`{Name: evConnectionClose, Src: []string{stateIdle, statePrepareUpdate, ....}, Dst: stateInit},`

Comment (mykola-kobets-epam) [reply umcontroller/umcontroller.go]: I missed that. changed

**Review by mlohvynenko — APPROVED** [review]

File: umcontroller/umcontroller.go [hunk 2]
```diff
@@ -1312,3 +1316,25 @@ func (umCtrl *Controller) closeUMServer() {
 
 	umCtrl.server.Stop()
 }
+
+func (umCtrl *Controller) allNodesConnected() bool {
```
Comment (al1img) [inline umcontroller/umcontroller.go:1320]: isAllNodesConnected

File: umcontroller/umcontroller.go [hunk 3]
```diff
@@ -1312,3 +1316,25 @@ func (umCtrl *Controller) closeUMServer() {
...
+	notConnectedNodeInd := slices.IndexFunc(umCtrl.connections, func(conn umConnection) bool {
+		return conn.handler == nil
+	})
+
+	return notConnectedNodeInd < 0
+}
+
+func (umCtrl *Controller) processAllNodesConnected() {
+	if umCtrl.fsm.Current() == stateInit {
```
Comment (al1img) [inline umcontroller/umcontroller.go:1329]: if umCtrl.fsm.Current() != stateInit {
    return
}

File: umcontroller/umcontroller.go [hunk 4]
```diff
@@ -1213,29 +1199,47 @@ func (umCtrl *Controller) createConnections() error {
 }
 
 func (umCtrl *Controller) handleNodeInfoChange(nodeInfo cloudprotocol.NodeInfo) {
-	if nodeInfo.Status == "unprovisioned" {
+	switch nodeInfo.Status {
+	case cloudprotocol.NodeStatusPaused:
 		return
-	}
+	case cloudprotocol.NodeStatusUnprovisioned:
```
Comment (al1img) [inline umcontroller/umcontroller.go:1205]: Empty line between switch cases

Comment (mykola-kobets-epam) [reply umcontroller/umcontroller.go:1206]: fixed

File: umcontroller/umcontroller.go (hunk 2 above)
Comment (mykola-kobets-epam) [reply umcontroller/umcontroller.go]: fixed

File: umcontroller/umcontroller.go (hunk 3 above)
Comment (mykola-kobets-epam) [reply umcontroller/umcontroller.go]: fixed

**Review by al1img — APPROVED** [review]

**Review by al1img — APPROVED** [review]

**Pull request merged**